certificate-rotation.md

```diff
@@ -0,0 +1,113 @@
+1. generate new CA certificate and combine it with the existing one
+1. push combined CA certificate to all nodes and reload certificates
+1. make combined CA available to all clients. We may need to wait some time (days/weeks) for all clients to make use of the new certificate
+1. generate and push new node/client certificates using the new CA certificate
+
+It is recommended to change CA certificates on a completely different schedule. eg: months before.
+**WARNING**: node certificates generated with the new CA will not be accepted by clients that do not yet have the new CA. Before all clients have the new CA cert, node certificates should be signed with the old CA (this is still a [TODO](https://github.com/cockroachdb/cockroach/blob/master/pkg/cli/cert.go#L91-L93))
+
+## Generating new CA certificates
+
+CA rotation includes generating a new CA certificate and key.
+
+If the original certificate and key are generated using:
+```
+$ cockroach cert create-ca --certs-dir=certs --ca-key=ca.key --logtostderr=true
+I170425 14:20:29.394473 1 security/certs.go:120  Generated CA key ca.key
+I170425 14:20:29.398081 1 security/certs.go:173  Wrote 1 certificates to certs/ca.crt
+```
+
+We can generate a new CA certificate and key with the following:
+```
+$ cockroach cert create-ca --certs-dir=certs --ca-key=ca.new.key --overwrite --logtostderr=true
+I170425 14:21:10.490341 1 security/certs.go:120  Generated CA key ca.new.key
+I170425 14:21:10.494007 1 security/certs.go:160  Found 1 certificates in certs/ca.crt
+I170425 14:21:10.494341 1 security/certs.go:173  Wrote 2 certificates to certs/ca.crt
+```
+
+Flags:
+* `--ca-key=ca.new.key`: path to the new key to generate. To re-use the existing key (not recommended), use `--ca-key=ca.key` and `--allow-key-reuse`.
+* `--overwrite`: add the new certificate to the existing `ca.crt` file.
+
+This results in a `ca.crt` containing the new certificate followed by the old certificate. This
+combined CA certificate should be used by all nodes and clients to verify old node/client certificates
+and well as new certificates.
+
+## Generating new node/client certificates
+
+Client and node certificates are simpler than CA certificates because we do not need to keep old
+certificates around.
+
+Generating a new certificate and key is the same as the original generation. If the files already exist, specify `--overwrite` to overwrite both certificate and key.
+
+If the original certificate and key for node `localhost` were created with:
+```
+$ cockroach cert create-node --certs-dir=certs --ca-key=ca.key --logtostderr=true localhost
+I170425 14:28:10.713448 1 security/certs.go:226  Generated node certificate: certs/node.crt
+I170425 14:28:10.713845 1 security/certs.go:232  Generated node key: certs/node.key
+```
+
+We can either delete the `node.crt` and `node.key` files and rerun the same command, or run:
+```
+roach cert create-node --certs-dir=certs --ca-key=ca.key --logtostderr=true --overwrite localhost
```

Comment:
@mberhault Shouldn't "ca.key" be replaced with "ca.new.key"?

Reply:
If we had renewed the CA certificate and key, then yes. However, CA certs and node/client certs can be regenerated on a different schedule (they have different expiration dates, with CA certs expiring much later).
For the purposes of the tutorial/docs, it may be better to change the CA key generation to:
1. rename the old CA key to `ca.old.key`
1. generate the new CA certificate and key: `ca.crt` and `ca.key`.

This way, the node/client certificate generation works regardless of whether we regenerated the CA.